aspnetcore/blazor/forms-validation.md

```diff
@@ -184,6 +243,33 @@ Create a component with the following markup, and use the component just as `Inp
         this, __value => CurrentValueAsString = __value, CurrentValueAsString)" />
 ```
 
+## Work with radio buttons
+
+Blazor doesn't provide a built-in scenario for handling radio button `<input>` elements. Radio button support is under consideration for a future Blazor release. The following workaround is recommended:
+
+```razor
+@foreach (var choice in new[] { Choices.Red, Choices.Green, Choices.Blue })
+{
+   <label>
+       <input name="yourColor" type="radio"
+            value="@choice"
+            checked="@(currentChoice == choice)"
+            @onchange="@(() => { currentChoice = choice; })" />
+       @choice.ToString()
+   </label>
+}
+
+<p>You chose: @currentChoice</p>
+
+@code {
+    private enum Choices { Red, Green, Blue };
+    private Choices currentChoice = Choices.Red;
+}
+```
+
+> [!NOTE]
+> The workaround shown in this section doesn't fire the `OnFieldChanged` form event that built-in `Input` components support.
```

Comment:
This is unrelated to data binding to radio buttons - the `OnFieldChanged` event is used for validation. You can still validate when using radio buttons. We don't provide you an out of the box `InputRadio` component, but you can implement one like this:

```razor
@using System.Globalization
@using System.ComponentModel.DataAnnotations
@typeparam TValue

@inherits InputBase<TValue>

<input @attributes="AdditionalAttributes" type="radio" value="@SelectedValue" checked="@(SelectedValue.Equals(Value))" @onchange="OnChange" />

@code {
    [Parameter]
    public TValue SelectedValue { get; set; }

    void OnChange(ChangeEventArgs args)
    {
        CurrentValueAsString = args.Value.ToString();
    }

    protected override bool TryParseValueFromString(string value, out TValue result, out string validationErrorMessage)
    {
        var success = BindConverter.TryConvertTo<TValue>(value, CultureInfo.CurrentCulture, out var parsedValue);
        if (success)
        {
            result = parsedValue;
            validationErrorMessage = null;
            return true;
        }
        else
        {
            result = default;
            validationErrorMessage = $"The {FieldIdentifier.FieldName} field is not valid.";
            return false;
        }
    }
}
```

And then here's how you use it:

```razor
<EditForm Model="model" OnValidSubmit="OnValidSubmit">
    <DataAnnotationsValidator />
    <ValidationSummary />
    @for (int i = 1; i <= 5; i++)
    {
        <label>
            <InputRadio name="yourColor" SelectedValue="i" @bind-Value="model.Rating" />
            @i
        </label>
    }
    <button type="submit">Submit</button>
</EditForm>

<p>You chose: @model.Rating</p>

@code {
    private Model model = new Model();

    void OnValidSubmit()
    {
        Console.WriteLine("valid");
    }

    public class Model
    {
        [Range(1, 5)]
        public int Rating { get; set; }
    }
}
```